internal/server/service.go

```diff
@@ -417,16 +418,19 @@ func (s *Service) createMiddleware(options ServiceOptions, certManager CertManag
 func (s *Service) serviceRequestWithTarget(w http.ResponseWriter, r *http.Request) {
 	LoggingRequestContext(r).Service = s.name
 
-	if s.shouldRedirectToHTTPS(r) {
-		s.redirectToHTTPS(w, r)
```

Comment:
As @AnyCPU mentioned, if we're no longer calling these two methods we should remove them.